### PR TITLE
Afford monthly recurrence

### PR DESCRIPTION
**Pull request by peaceful-james**
This is just a small bit of work done off the PR by @Stroemgren.
I rebased that work off current master to get a mergeable PR, for expediency.
Hopefully it's enough to get merged because monthly recurrence would be 🔥

### CODE REVIEW

File: lib/cocktail/parser/i_calendar.ex [hunk 1]
```diff
@@ -232,6 +239,20 @@ defmodule Cocktail.Parser.ICalendar do
...
+  defp parse_mday_string(mday_string) do
+    case Integer.parse(mday_string) do
+      {mday, ""} ->
+        if mday in -31..31, do: {:ok, mday}, else: {:error, :invalid_mday}
+
+      {_mday, _remainder} ->
+        {:error, :invalid_mday}
+
+      :error ->
+        {:error, :invalid_mday}
```
Comment (peaceful-james) [inline lib/cocktail/parser/i_calendar.ex:252]: Could be reduced to single "catch all" case:
```
_anything ->
{:error, :invalid_mday}
```

File: lib/cocktail/validation/schedule_lock.ex [hunk 2]
```diff
@@ -31,4 +31,34 @@ defmodule Cocktail.Validation.ScheduleLock do
...
+          start_time_day_of_month ->
+            next_month_date = Timex.shift(time, months: 1)
+            # Timex.set already handle the marginal case like setting a day of month more than the month contains
+            next_month_date
+            |> Timex.set(day: start_time_day_of_month)
+            |> Timex.diff(time, :days)
+        end
+
+      shift_by(day_diff, :days, time)
+    end
```
Comment (peaceful-james) [inline lib/cocktail/validation/schedule_lock.ex:62]: This large if expression is quite tasteless. Any suggestions for a nice way to write it?

**Comment by peaceful-james** [comment]
One thing I cannot figure out is how to do "last Friday of every month".

**Comment by peaceful-james** [comment]
I opened this PR too soon. I just realised iCalendar string creation is not being done correctly. Tests to follow.

**Comment by peaceful-james** [comment]
@doughsay I think this PR is good to go now. Are you the right person to poke for a review?

**Comment by doughsay** [comment]
Yes I am, sorry I've been busy. Will look later today...

**Comment by peaceful-james** [comment]
> Yes I am, sorry I've been busy. Will look later today...

No rush at all. I am using my own fork of it for the time being.

Let me know if I can help or improve it in any way. I know this kind of PR can be trying to review.

**Comment by doughsay** [comment]
Thanks for all the work on this @peaceful-james and for the original work by @bruteforcecat and @Stroemgren! Even if this doesn't follow the iCal spec perfectly, I think we're much closer than we were, and any other issue can be worked out over time.  

The CI appears to be broken, and I'm going to update to using github actions anyway, so I'm going to ignore that for now.  I'm going to merge this as is, and do a little cleanup before doing a new release.

**Review by doughsay — APPROVED** [review]

**Pull request merged**